src/lib.rs

```diff
@@ -400,6 +403,18 @@ impl<'tree, Tab> DockArea<'tree, Tab> {
                                     Sense::click_and_drag()
                                 };
 
+                                response.0.clone().context_menu(|ui| {
+                                    tab_viewer.context_menu(ui, tab);
+                                    if style.show_close_buttons && ui.button("close").clicked() {
```

Comment:
Replace `button` with `menu_button`, capitalise the word "Close" in the string.

Also add `ui.close_menu()` at the end of this `if`.

Reply:
`menu_button` is for top level buttons only [example](https://docs.rs/egui/0.19.0/egui/menu/index.html)
and inside menus there normal buttons

Reply:
`ui.close_menu()` at the end of this if would close the menu immediately

Reply:
Ok you're right